packages/labs/signals/src/test/signal-watcher_test.ts

```diff
@@ -0,0 +1,230 @@
+    class TestElement extends SignalWatcher(LitElement) {
+      override render() {
+        return html`<p>count: ${count.get()}</p>`;
+      }
+    }
+    customElements.define(generateElementName(), TestElement);
+    const el = new TestElement();
+    container.append(el);
+
+    await el.updateComplete;
+    assert.equal(el.shadowRoot?.querySelector('p')?.textContent, 'count: 0');
+
+    count.set(1);
+    // await new Promise((r) => setTimeout(r, 0));
+    await el.updateComplete;
+    assert.equal(el.shadowRoot?.querySelector('p')?.textContent, 'count: 1');
+  });
+
+  test('non-signal updates work', async () => {
+    const count = new Signal.State(0);
+    let renderCount = 0;
+    class TestElement extends SignalWatcher(LitElement) {
+      @property()
+      foo = 'foo';
+
+      override render() {
+        renderCount++;
+        return html`
+          <p>count: ${count.get()}</p>
+          <p>foo: ${this.foo}</p>
+        `;
+      }
+    }
+    customElements.define(generateElementName(), TestElement);
+    const el = new TestElement();
+    container.append(el);
+
+    await el.updateComplete;
+    const p1 = el.shadowRoot!.querySelectorAll('p')[0]!;
+    const p2 = el.shadowRoot!.querySelectorAll('p')[1]!;
+
+    assert.equal(p1.textContent, 'count: 0');
+    assert.equal(p2.textContent, 'foo: foo');
+    assert.equal(renderCount, 1);
+
+    count.set(1);
+    await el.updateComplete;
+    assert.equal(p1.textContent, 'count: 1');
+    assert.equal(p2.textContent, 'foo: foo');
+    assert.equal(renderCount, 2);
+
+    el.foo = 'bar';
+    await el.updateComplete;
+    assert.equal(p1.textContent, 'count: 1');
+    assert.equal(p2.textContent, 'foo: bar');
+    assert.equal(renderCount, 3);
+
+    count.set(2);
+    el.foo = 'baz';
+    await el.updateComplete;
+    assert.equal(p1.textContent, 'count: 2');
+    assert.equal(p2.textContent, 'foo: baz');
+    assert.equal(renderCount, 4);
+  });
+
+  test('unsubscribes to a signal on element disconnect', async () => {
+    let readCount = 0;
+    const count = new Signal.State(0);
+    const countPlusOne = new Signal.Computed(() => {
+      readCount++;
+      return count.get() + 1;
+    });
+
+    class TestElement extends SignalWatcher(LitElement) {
+      override render() {
+        return html`<p>count: ${countPlusOne.get()}</p>`;
+      }
+    }
+    customElements.define(generateElementName(), TestElement);
+    const el = new TestElement();
+    container.append(el);
+
+    // First render, expect one read of the signal
+    await el.updateComplete;
+    assert.equal(
+      el.shadowRoot?.querySelector('p')?.textContent,
+      'count: 1',
+      'A'
+    );
+    assert.equal(readCount, 1);
+
+    // Updates work
+    count.set(1);
+    await el.updateComplete;
+    assert.equal(
+      el.shadowRoot?.querySelector('p')?.textContent,
+      'count: 2',
+      'B'
+    );
+    assert.equal(readCount, 2);
+
+    // Disconnect the element
```

Comment:
Add a test like this: https://github.com/lit/lit/pull/4707/files#diff-9e4d56fda3753440f204ca2f7f5fbaad99a1809237a396a74a28e7b80034a7a7R106

Reply:
Just a note to myself here that I didn't add this test yet.